### PR TITLE
Implement large file streaming for proxy GET

### PR DESCRIPTION
**Pull request by jdcasey**
This introduces another StreamingOutput implementation that wraps a Buffer response coming from Mutiny when we proxy a GET request for a file. This enables us to avoid buffering the whole file in memory when transferring back to the client. This is based on work in Commonjava/gateway.

**This code has NOT been tested for performance, to ensure that the memory usage problems are fixed. We need to performance test this, probably with a profiler on localhost or similar at a minimum, to ensure heap memory doesn't spike when a really large file is proxied.**

NOTE: This is a backport of the attempt to implement file streaming on the archive-enabled version of this service. I've cut out digest handling and tracking record setup to make it work in the older codebase. It's based on commit: 989c6c165741d0ec20d7b6cf8611187a9e3a9adb.

### CODE REVIEW

File: src/main/java/org/commonjava/util/sidecar/util/BufferStreamingOutput.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            int total = buffer.length();
+            int transferred = 0;
+            while ( transferred < total )
+            {
+                int next = bufSize < total ? bufSize : total;
+                byte[] bytes = buffer.getBytes( transferred, next );
+                out.write( bytes );
+
+                transferred = next;
+            }
```
Comment (ligangty) [inline src/main/java/org/commonjava/util/sidecar/util/BufferStreamingOutput.java:63]: Will this cause some content missing? For example, when transferred == total(Means reached the end of the buffer), this while loop will end, but the left part of the transferred is not write to the outputstream(not trigger the out.write(bytes) for last time).

File: src/main/java/org/commonjava/util/sidecar/util/BufferStreamingOutput.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                    out = new TeeOutputStream( cacheStream, output );
+                }
+            }
+
+            Buffer buffer = response.bodyAsBuffer();
+            int total = buffer.length();
+            int transferred = 0;
+            while ( transferred < total )
+            {
+                int next = bufSize < total ? bufSize : total;
```
Comment (ligangty) [inline src/main/java/org/commonjava/util/sidecar/util/BufferStreamingOutput.java:58]: Maybe this can be moved out of the while loop as bufSize and total will never change? And maybe Math.min(bufSize, total) is more readable?

File: src/main/java/org/commonjava/util/sidecar/util/BufferStreamingOutput.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                byte[] bytes = buffer.getBytes( transferred, next );
+                out.write( bytes );
+
+                transferred = next;
```
Comment (ligangty) [inline src/main/java/org/commonjava/util/sidecar/util/BufferStreamingOutput.java:62]: I'm thinking the quit condition here for the while loop. There are two cases:
1. For small files whose size < bufSize, so there will be only one loop and can quit successfully.
2. For large files whose size > bufSize. In this case, the "total" will be the real file size(from the buffer.length), right? So that means the transferred will always be set to bufSize because of the "bufSize<total". Will this cause a dead loop as transferred will be less than total forever?
